src/meta-srv/src/metasrv/builder.rs

```diff
@@ -129,13 +135,33 @@ impl MetaSrvBuilder {
 
         let selector = selector.unwrap_or_else(|| Arc::new(LeaseBasedSelector));
 
+        let pushers = Arc::new(RwLock::new(BTreeMap::new()));
```

Comment:
The `pushers` may be wrapped with a struct `Pushers`:

```rust
struct Pushers {
   m: RwLock::new(BTreeMap::new()),
}
type PushersRef = Arc<Pushers>;
```

And move its logic into its `impl`. I think the code will be more clear.